test/e2e/operator.go

```diff
@@ -0,0 +1,83 @@
+	},
+	"buildah": {
+		ClusterBuildStrategy: "samples/buildstrategy/buildah/buildstrategy_buildah_cr.yaml",
+		Build:                "samples/build/build_buildah_cr.yaml",
+		BuildRun:             "samples/buildrun/buildrun_buildah_cr.yaml",
+	},
+	"buildpacks-v3": {
+		ClusterBuildStrategy: "samples/buildstrategy/buildpacks-v3/buildstrategy_buildpacks-v3_cr.yaml",
+		Build:                "samples/build/build_buildpacks-v3_cr.yaml",
+		BuildRun:             "samples/buildrun/buildrun_buildpacks-v3_cr.yaml",
+	},
+	"buildpacks-v3-namespaced": {
+		BuildStrategy: "samples/buildstrategy/buildpacks-v3/buildstrategy_buildpacks-v3_namespaced_cr.yaml",
+		Build:         "samples/build/build_buildpacks-v3_namespaced_cr.yaml",
+		BuildRun:      "samples/buildrun/buildrun_buildpacks-v3_namespaced_cr.yaml",
+	},
+}
+
+// privateTestCases contains all test cases design to run against private git repositories, those
+// tests are enabled via environment variable ("TEST_PRIVATE_REPO").
+var privateTestCases = map[string]*SampleFiles{
+	"private-github-kaniko": {
+		ClusterBuildStrategy: "samples/buildstrategy/kaniko/buildstrategy_kaniko_cr.yaml",
+		Build:                "test/data/build_kaniko_cr_private_github.yaml",
+		BuildRun:             "samples/buildrun/buildrun_kaniko_cr.yaml",
+	},
+	"private-gitlab-kaniko": {
+		ClusterBuildStrategy: "samples/buildstrategy/kaniko/buildstrategy_kaniko_cr.yaml",
+		Build:                "test/data/build_kaniko_cr_private_gitlab.yaml",
+		BuildRun:             "samples/buildrun/buildrun_kaniko_cr.yaml",
+	},
+	"private-github-buildah": {
+		ClusterBuildStrategy: "samples/buildstrategy/buildah/buildstrategy_buildah_cr.yaml",
+		Build:                "test/data/build_buildah_cr_private_github.yaml",
+		BuildRun:             "samples/buildrun/buildrun_buildah_cr.yaml",
+	},
+	"private-gitlab-buildah": {
+		ClusterBuildStrategy: "samples/buildstrategy/buildah/buildstrategy_buildah_cr.yaml",
+		Build:                "test/data/build_buildah_cr_private_gitlab.yaml",
+		BuildRun:             "samples/buildrun/buildrun_buildah_cr.yaml",
+	},
+	"private-github-buildpacks-v3": {
+		ClusterBuildStrategy: "samples/buildstrategy/buildpacks-v3/buildstrategy_buildpacks-v3_cr.yaml",
+		Build:                "test/data/build_buildpacks-v3_cr_private_github.yaml",
+		BuildRun:             "samples/buildrun/buildrun_buildpacks-v3_cr.yaml",
+	},
+	"private-github-s2i": {
+		ClusterBuildStrategy: "samples/buildstrategy/source-to-image/buildstrategy_source-to-image_cr.yaml",
+		Build:                "test/data/build_source-to-image_cr_private_github.yaml",
+		BuildRun:             "samples/buildrun/buildrun_source-to-image_cr.yaml",
+	},
+}
+
+// OperatorTests execute test cases.
+func OperatorTests(t *testing.T, ctx *framework.TestCtx, f *framework.Framework) {
+	samplesTesting := NewSamplesTesting(t, ctx, f)
+	if os.Getenv(EnvVarEnablePrivateRepos) == "true" {
+		samplesTesting.TestAll(privateTestCases)
```

Comment:
Does not this code mean that when the private repositories environment variable is set to `true`, that it **only** runs the private repositories test cases? The old behavior imo was to run them **in addition** to those on public repositories.

Reply:
Yes, that's correct, currently would be one or the other. You have a good point, will make sure both are supported.